foreshadow/foreshadow.py

```diff
@@ -311,6 +316,110 @@ def score(self, data_df, y_df=None, sample_weight=None):
         self._prepare_predict(data_df.columns)
         return self.pipeline.score(data_df, y_df, sample_weight)
 
+    def dict_serialize(self, deep=True):
+        """Serialize the init parameters of the foreshaow object.
+
+        Args:
+            deep (bool): If True, will return the parameters for this estimator
+                recursively
+
+        Returns:
+            dict: The initialization parameters of the foreshadow object.
+
+        """
+        params = self.get_params(deep)
+        selected_params = self.customize_serialization_params(params)
+        serialized = _make_serializable(
+            selected_params, serialize_args=self.serialize_params
+        )
+        serialized["estimator"] = self.__customize_serialized_estimator(
+            self.estimator
+        )
+        return serialized
+
+    @staticmethod
+    def __customize_serialized_estimator(estimator):
+        result = dict()
+        is_meta_estimator = False
+        if isinstance(estimator, MetaEstimator):
+            estimator = estimator.estimator
+            is_meta_estimator = True
+
+        serialized_estimator = estimator.get_params()
+        serialized_estimator["_class"] = (
+            estimator.__module__ + "." + type(estimator).__name__
+        )
+        serialized_estimator["_method"] = "dict"
+
+        if is_meta_estimator:
+            result["estimator"] = serialized_estimator
+            result["preprocessor"] = "_y_preparer"
+            result["_class"] = "MetaEstimator"
+            result["_method"] = "dict"
+        else:
+            result = serialized_estimator
+        return result
+
+    @classmethod
+    def dict_deserialize(cls, data):
+        """Deserialize the dictionary form of a foreshadow object.
+
+        Args:
+            data: The dictionary to parse as foreshadow object is constructed.
+
+        Returns:
+            object: A re-constructed foreshadow object.
+
+        """
+        serialized_estimator = data.pop("estimator")
+        is_meta_estimator = "MetaEstimator" == serialized_estimator["_class"]
+        estimator = cls.__reconstruct_estimator(
+            serialized_estimator, is_meta_estimator
+        )
+
+        params = _make_deserializable(data)
+        data_columns = params.pop("data_columns")
+        params["estimator"] = estimator
+
+        ret_tf = cls(**params)
+        ret_tf.data_columns = data_columns
+        if is_meta_estimator:
+            ret_tf.estimator = MetaEstimator(
+                estimator=estimator, preprocessor=ret_tf.y_preparer
+            )
+        return ret_tf
+
+    @classmethod
+    def __reconstruct_estimator(cls, data, is_meta_estimator):
+        # TODO untested on non-meta-estimator
+        if is_meta_estimator:
```

Comment:
why not just make this the default and not use this flag?

Reply:
This argument won't be necessary in the latest change.